src/components/ui/Icon/index.js

```diff
@@ -0,0 +1,250 @@
+    'stroke-width': featherStrokeWidth,
+    // Get the rest of the needed Feather attributes.
+    ...otherFeatherAttrs
+  } = Feather.icons[name].attrs;
+
+  const { hasContainer, padding, size, strokeWidth } = defineIconSizes({
+    ...props,
+    theme,
+  });
+
+  return (
+    <span className={className} data-testid={dataTestId}>
+      <svg
+        {...otherFeatherAttrs}
+        // These props are placed above the {...otherProps} spread so
+        // user-supplied props can override our default values.
+        role="img"
+        strokeLinecap={strokeLinecap}
+        strokeLinejoin={strokeLinejoin}
+        strokeWidth={strokeWidth}
+        {...otherProps}
+        aria-hidden={title ? undefined : true}
+        aria-labelledby={title && titleId}
+        id={svgId}
+        css={css`
+          fill: ${fillColor};
+          height: ${toUnits(size)};
+          stroke: ${strokeColor};
+          stroke-width: ${toUnits(strokeWidth)};
+          width: ${toUnits(size)};
+
+          ${hasContainer &&
+            css`
+              margin: ${toUnits(padding)};
+            `}
+
+          ${!fillColor &&
+            !strokeColor &&
+            // If no explicit colour was specified, we drop the opacity to
+            // simulate lowering the intensity of the icon's colour.
+            css`
+              stroke: currentColor;
+              opacity: 0.8;
+            `}
+        `}
+      >
+        {title && <title id={titleId}>{title}</title>}
+        {description && <desc id={descriptionId}>{description}</desc>}
+        <g
+          // eslint-disable-next-line react/no-danger
+          dangerouslySetInnerHTML={{ __html: Feather.icons[name].contents }}
+        />
+        {children}
+      </svg>
+    </span>
+  );
+};
+
+export const styles = (props) => {
+  const { border, background, margin, verticalAlign } = props;
+  const { borderWidth, wrapperSize } = defineIconSizes(props);
+  let borderBackground;
+  let backgroundCSS;
+  let borderCSS;
+
+  if (border) {
+    borderCSS = css`
+      border: ${toUnits(borderWidth)} solid ${border};
+    `;
+  }
+
+  if (background) {
+    backgroundCSS = css`
+      background: ${background};
+    `;
+  }
+
+  if (border || background) {
+    borderBackground = css`
+      ${backgroundCSS};
+      ${borderCSS};
+      border-radius: 50%;
+      height: ${toUnits(wrapperSize)};
+      width: ${toUnits(wrapperSize)};
+    `;
+  }
+
+  return css`
+    ${borderBackground};
+    display: inline-block;
+    margin: ${margin};
+    vertical-align: ${verticalAlign};
+  `;
+};
+
+Icon.defaultProps = {
+  children: undefined,
+  className: undefined,
+  'data-testid': undefined,
+  description: undefined,
+  id: undefined,
+  fillColor: undefined,
+  strokeColor: undefined,
+  title: undefined,
+  extraSmall: false,
+  small: false,
+  medium: true,
+  large: false,
+  extraLarge: false,
+  margin: '0 0.8rem 0 0', // @todo this default should probably just pull from the theme padding!
```

Comment:
I think just `TODO:` will work. Also: is this something you could do before we merge? You could set this as `undefined` or `null` by default, check for it being `undefined` in the component, and then set it based on a theme value if so. I can do that for you; what should the value be based on in the theme?

Reply:
I wasn't sure if this was the right approach, or if there were a way of pulling theme values directly into the React default props. I'm okay with putting our actual defaults in the component code though (and I've done this elsewhere!) so I've implemented this now.

Reply:
There's actually no way to add the `theme` defaults in the `defaultProps` because `defaultProps` are defined before the `theme` is passed to the component. Essentially we default to "using values from the theme", but that can't be a default prop because we don't yet _have_ the theme. So your approach is exactly the right one! 👍